doc/contributing/adding-new-napi-api.md

```diff
@@ -1,66 +1,60 @@
-introduce modifications to already-released Node-APIs, as long as the
-modifications affect neither the ABI nor the API of earlier versions. Such
-modifications **must** be accompanied by an opt-out flag. This provides add-on
-maintainers who take advantage of the initial compile-time flag to track
-impending changes to Node-API with
+1. **Adherence to Node-API standards**
+   * **Must** be a C API.
+   * **Must** not throw exceptions.
+   * **Must** return `napi_status`.
+   * **Should** consume `napi_env`.
+   * **Must** operate on primitive data types, pointers to primitive data types, or opaque handles.
+   * **Must** be a necessary API, not a convenience API (which belongs in node-addon-api).
+   * **Must** not break ABI compatibility with other Node.js versions.
 
-* a quick fix to the breakage caused,
-* a notification that such breakage is impending, and thus
-* a buffer to adoption above and beyond the one provided by the initial
-  compile-time flag.
+2. **Maintaining VM agnosticism**
+   * New APIs **should** be compatible with various JavaScript VMs.
+
+## Documentation and testing
+
+1. **Documentation**
+   * PRs introducing new APIs **must** include corresponding documentation updates.
+   * Experimental APIs **must** be clearly documented as experimental and require an explicit compile-time flag
+     to opt-in (`#define`).
+
+2. **Testing**
+   * PRs **must** include at least one test case demonstrating API usage.
+   * **Should** include test cases for various interesting uses of the API.
+   * **Should** provide a sample demonstrating realistic usage, similar to a real-world addon.
+
+## Process and approval
+
+1. **Team discussion**
+   * New APIs **should** be discussed in a Node-API team meeting.
+
+2. **Review and approval**
+   * A new API addition **must** be signed off by at least two Node-API team members.
+   * **Should** be implemented in at least one other VM implementation of Node.js.
+
+3. **Experimental phase**
+   * New APIs **must** be marked as experimental for at least one minor Node.js release before promotion.
+   * **Must** have a feature flag (`NODE_API_EXPERIMENTAL_HAS_<FEATURE>`) for distinguishing experimental features.
+   * **Must** be considered for backporting.
+   * Exit criteria from experimental status include:
+     * Opening a PR in `nodejs/node` to remove experimental status, tagged as **node-api** and **semver-minor**.
+     * Approval by the Node-API team.
+     * Availability of a down-level implementation if backporting is needed.
+     * Usage by a published real-world module to support de-experimentation.
+     * Implementation in a Node.js version with an alternate VM.
+
+## Modifying released APIs
+
+Changes to existing APIs that do not affect ABI or API compatibility with earlier versions are permissible
+with an opt-out flag (`NODE_API_EXPERIMENTAL_<FEATURE>_OPT_OUT`). This helps addon maintainers by providing:
```

Comment:
I don't think `NODE_API_EXPERIMENTAL_<FEATURE>_OPT_OUT` exists in the original paragraph.

Reply:
Ha! I have no idea how I put that in there, sometimes I just blank, weird! I'll fix it.

Reply:
Actually, looking over the docs, this flag does exist https://github.com/nodejs/node/blob/7430638df8ce347ee2a53de6a734948e848ab6c1/doc/api/n-api.md?plain=1#L246

Reply:
This is not a requirement though. Bullet points added in this document would be treated as a requirement.